### PR TITLE
Fix FP S1226 (`no-parameter-reassignment`): Allow inside if statements

### LINKED ISSUE #3364: Fix FP S1226 (`no-parameter-reassignment`): Why only on top-level `if`?
The rule is currently just ignoring cases on top level if statements inside a function. Probably we should normalize for cases inside any conditional expressions?

Reported in [community](https://community.sonarsource.com/t/fp-javascript-s1226-sonar-sometimes-doesnt-recognize-that-param-reassignment-is-conditional/70545)

### PR DESCRIPTION
**Pull request by victor-diez-sonarsource**
Fixes #3364

### CODE REVIEW

**Review by ilia-kebets-sonarsource — CHANGES_REQUESTED** [review]
Can you add a test case and fix the checks, please?

**Comment by victor-diez-sonarsource** [comment]
RSPEC update: https://github.com/SonarSource/rspec/pull/1322

File: eslint-bridge/src/linting/eslint/rules/no-parameter-reassignment.ts [hunk 1]
```diff
@@ -228,26 +228,20 @@ export const rule: Rule.RuleModule = {
...
-      hasParentOfType(ifStatementParent.parent, ['BlockStatement']) &&
-      hasParentOfType(ifStatementParent.parent.parent, FUNCTION_NODES)
-    );
+function isInsideIfStatement(context: Rule.RuleContext) {
+  const ancestors = context.getAncestors();
+  for (let i = ancestors.length - 1; i >= 0; i--) {
+    if (
+      ancestors[i].type === 'IfStatement' &&
+      i < ancestors.length - 1 &&
+      ancestors[i + 1] === (ancestors[i] as estree.IfStatement).consequent
```
Comment (ilia-kebets-sonarsource) [inline eslint-bridge/src/linting/eslint/rules/no-parameter-reassignment.ts:237]: WDYT about defining a method:

```javascript
function isConsequentTo(ancestors, index) {
  return index < ancestors.length - 1 &&
  ancestors[index + 1] === (ancestors[index] as estree.IfStatement).consequent
}
```

**Review by ilia-kebets-sonarsource — APPROVED** [review]
LGTM, I just left a cosmetic proposal.

**Pull request merged**